core/src/executor/aggregate/state.rs

```diff
@@ -175,39 +177,87 @@ impl<'a> State<'a> {
+                None => Err(AggregateError::ValueNotFound(ident.to_string()).into()),
+            },
             Expr::CompoundIdentifier(idents) => {
                 if idents.len() != 2 {
-                    return Err(AggregateError::UnsupportedCompoundIdentifier(expr.clone()));
+                    return Err(AggregateError::UnsupportedCompoundIdentifier(expr.clone()).into());
                 }
 
                 let table_alias = &idents[0];
                 let column = &idents[1];
 
-                context
-                    .get_alias_value(table_alias, column)
-                    .ok_or_else(|| AggregateError::ValueNotFound(column.to_string()))
+                match context.get_alias_value(table_alias, column) {
+                    Some(x) => Ok(x),
+                    None => Err(AggregateError::ValueNotFound(column.to_string()).into()),
+                }
             }
-            _ => Err(AggregateError::OnlyIdentifierAllowed),
-        };
+            _ => Err(AggregateError::OnlyIdentifierAllowed.into()),
+        }
+    }
+
+    fn accumulate_get_value(&self, context: &BlendContext<'_>, expr: &Expr) -> Result<Value> {
+        match expr {
+            Expr::Identifier(_ident) => match self.accumulate_get_value1(context, expr) {
+                Ok(x) => {
+                    let y: Value = x.to_owned();
+                    Ok(y)
+                }
+                Err(x) => Err(x),
+            },
+            Expr::CompoundIdentifier(_idents) => match self.accumulate_get_value1(context, expr) {
+                Ok(x) => {
+                    let y: Value = x.to_owned();
+                    Ok(y)
+                }
+                Err(x) => Err(x),
+            },
+            Expr::Literal(x) => match x {
+                AstLiteral::Number(x) => match Decimal::from_str(&x.to_string()) {
+                    Ok(x) => Ok(Value::Decimal(x)),
+                    _ => Err(AggregateError::UnsupportedAggregateLiteral.into()),
+                },
+                _ => Err(AggregateError::UnsupportedAggregateLiteral.into()),
+            },
+            Expr::BinaryOp { left, op, right } => {
+                let left_value: &Value = &self.accumulate_get_value(context, left)?;
+                let right_value: &Value = &self.accumulate_get_value(context, right)?;
 
+                match op {
+                    BinaryOperator::Plus => left_value.add(right_value),
+                    BinaryOperator::Minus => left_value.subtract(right_value),
+                    BinaryOperator::Multiply => left_value.multiply(right_value),
+                    BinaryOperator::Divide => left_value.divide(right_value),
+                    BinaryOperator::Modulo => left_value.modulo(right_value),
+                    _ => Err(AggregateError::UnsupportedAggregateFunction.into()),
+                }
+            }
+            _ => Err(AggregateError::UnsupportedAggregateFunction.into()),
+        }
+    }
```

Comment:
Aggregate evaluation is not supposed to be done in here.
You may need to use
https://github.com/gluesql/gluesql/blob/e4c04c1135680b6199b965d732591029f46d808e/core/src/executor/evaluate/mod.rs#L27-L33

Reply:
> so we will need to supply a storage variable,
yes I think so. Corresponding interface change would be required for `aggregate/state.rs`